components/UserList.js

```diff
@@ -11,6 +14,69 @@ export default function UserList(props) {
     value: 'hidden',
   }
 
+  const colorArray = [
+    'bg-[#8DD5F3]',
+    'bg-[#FFABAB]',
+    'bg-[#A7F5A6]',
+    'bg-[#FFC672]',
+    'bg-[#D197FF]',
+    'bg-[#D2FC00]',
+  ]
+
+  function randomColor() {
+    const idx = Math.floor(Math.random() * colorArray.length)
+    return colorArray[idx]
+  }
+
+  const initialState = []
+
+  function setInitialState() {
+    props.userList.map((player) => {
+      if (props.currPlayer?.id !== player.id) {
+        initialState.push({ id: player.id, color: randomColor() })
+      }
+    })
+  }
+
+  useEffect(() => {
+    if (userColor == undefined) {
+      setInitialState()
+      setUserColor(initialState)
+    } else {
+      // new colors will be assigned as users join
+      if (userColor.length + 1 !== props.userList.length) {
+        setInitialState()
+        setUserColor(initialState)
+      }
+    }
+  }, [props.userList])
+
+  const [userColor, setUserColor] = useState()
+
+  // Returns true if user is in the array otherwise undefined
+  function isUserInState(userId) {
+    if (userColor == undefined) {
+      return false
+    }
+    return userColor.find((user) => {
+      return user.id === userId
+    })
+  }
+
+  // returns user color or a randomColor is the user isn't on the state
+  function getUserColorById(userId) {
+    if (isUserInState(userId)) {
+      return userColor.find((user) => {
+        return user.id == userId
+      }).color
```

Comment:
IsUserInState does the same thing as userColor.find.

It is easier to read in one function
return userColor.find((user) => {
 return user.id == userId
})?.color

Reply:
not exactly. if user isn't in the state it wont assign a color